http/request.go

```diff
@@ -167,29 +180,74 @@ func (r *Request) createHTTPRequest() (*http.Request, error) {
 }
 
 // sendRequest sends the request using the given HTTP client
-func (r *Request) sendRequest(client *http.Client, logger logger.Logger, retriesRemaining uint) (*Response, error) {
-	request, err := r.createHTTPRequest()
-	if err != nil {
+func (r *Request) sendRequest(ctx context.Context, client *http.Client, logger logger.Logger, retriesRemaining uint) (*Response, error) {
+	ctx, span := tracer.Start(ctx, r.url.Hostname()) // TODO: Decide on how to name this span
+	defer span.End()
+
+	span.SetAttributes(attribute.String("method", r.verb))
```

Comment:
Setting ottributes and errors should be part of the ctx e.g. `ctx.SetAttribute(key, val)` and `ctx.Errorf` e.g. https://github.com/flanksource/incident-commander/commit/58e771879d7464e0c7924dfdd8d68a25160dc726